app/models/calculator_resolver.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class CalculatorResolver
+  def self.call(calculator)
+    dependencies = {}
+    get_calculations(calculator).each do |field|
+      dependencies[field] = if field[:type] == 'Calculation'
+                              get_dependencies(field[:value])
+                            else
+                              []
+                            end
+    end
+
+    dependencies
+  end
+
+  def self.get_dependencies(value)
+    calculator = Dentaku::Calculator.new
+    calculator.dependencies(value)
```

Comment:
This returns array of string, not result requested in the ticket

Reply:
Thank you for response. I've rework the method to return the array of objects.